pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -192,6 +192,7 @@ public void findEntryFailed(ManagedLedgerException exception, Optional<Position>
             log.warn("[{}][{}] read failed from ledger at position:{} : {}", topicName, subName, failedReadPosition,
                     exception.getMessage());
             findEntryComplete(failedReadPosition.get(), ctx);
+            return;
```

Comment:
If we return here, how the `expirationCheckInProgress` will change to FALSE?
If the `expirationCheckInProgress` keep TRUE here. The subsequent message expiration task will always be skipped https://github.com/apache/pulsar/pull/17816/files#diff-813cddfda73ce8df0722a7c83ac1cb8d6d3db62ea13b44b5dc1bc438010fdd40R72

Reply:
Thanks for reviewing. The findEntryComplete method will change the `expirationCheckInProgress` to FALSE. If return is not added, the updateRates method will execute repeatedly

Reply:
> The findEntryComplete method will change the expirationCheckInProgress to FALSE. see line182

Please see line 107. If we return here, the expirationCheckInProgress=true, we will not have a chance to call `findEntryComplete` method